pkg/feature/resources.go

```diff
@@ -12,6 +15,21 @@ import (
 // This way we ensure that when the feature is cleaned up, the namespace will be deleted as well.
 func CreateNamespace(namespace string) Action {
 	return func(f *Feature) error {
+		// Despite the cluster.CreateNamespace function already checks if the target
+		// namespace exists, it seems relevant to do the check here. Otherwise, we may
+		// set or change the owner reference of an existent namespace, and that would lead
+		// to namespace deletion for cases where it is better to not terminate it.
+		foundNamespace := &corev1.Namespace{}
+		err := f.Client.Get(context.TODO(), client.ObjectKey{Name: namespace}, foundNamespace)
+		if err != nil {
+			if !apierrs.IsNotFound(err) {
+				return err
+			}
+		} else {
+			// Namespace exists. We do no-op.
```

Comment:
hmm, shouldn't the NS be deleted if found?

Reply:
I think we should not touch namespaces created by somebody else.

Reply:
so maybe update the comment before where it is stated that the NS is guaranteed to be clean. :)

```
// This way we ensure that when the feature is cleaned up, the namespace will be deleted as well.
```

Reply:
Fixed.